client/my-sites/hosting/php-version-card/index.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import CardHeading from 'components/card-heading';
+import MaterialIcon from 'components/material-icon';
+import Button from 'components/button';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getHttpData, requestHttpData } from 'state/data-layer/http-data';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import Spinner from 'components/spinner';
+import FormRadiosBar from 'components/forms/form-radios-bar';
+import { errorNotice, successNotice } from 'state/notices/actions';
+
+const requestId = ( siteId, method ) => `hosting-php-version-${ method }-${ siteId }`;
+
+export const requestPhpVersion = siteId => {
+	const method = 'GET';
+
+	return requestHttpData(
+		requestId( siteId, method ),
+		http(
+			{
+				method,
+				path: `/sites/${ siteId }/hosting/php-version`,
+				apiNamespace: 'wpcom/v2',
+				body: {},
+			},
+			{}
+		),
+		{
+			fromApi: () => version => {
+				return [ [ requestId( siteId, method ), version ] ];
+			},
+			freshness: 0,
+		}
+	);
+};
+
+export const setPhpVersion = ( siteId, version ) => {
+	const method = 'POST';
+
+	return requestHttpData(
+		requestId( siteId, method ),
+		http(
+			{
+				method,
+				path: `/sites/${ siteId }/hosting/php-version`,
+				apiNamespace: 'wpcom/v2',
+				body: {
+					version: version,
+				},
+			},
+			{}
+		),
+		{
+			fromApi: () => success => {
+				if ( true === success ) {
+					requestPhpVersion( siteId );
+				}
+				return [ [ requestId( siteId, method ), success ] ];
+			},
+			freshness: 0,
+		}
+	);
+};
+
+class PhpVersionCard extends React.Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			selectedPhpVersion: null,
+		};
+	}
+
+	componentDidMount() {
```

Comment:
Non-Blocking, but opportunity to use to learn about use https://reactjs.org/docs/hooks-effect.html, @jeraldjuice or @glendaviesnz could also give pointers!

Reply:
Is this the preferred way now?

Reply:
In general for React has a preference to move away from classes when we don't need it, but it's not blocking.